source/common/common/regex.cc

```diff
@@ -48,7 +48,7 @@ CompiledGoogleReMatcher::CompiledGoogleReMatcher(
   if (config.google_re2().has_max_program_size()) {
     const uint32_t max_program_size =
         PROTOBUF_GET_WRAPPED_OR_DEFAULT(config.google_re2(), max_program_size, 100);
-    if (regex_program_size > max_program_size) {
+    if (max_program_size > 0 && regex_program_size > max_program_size) {
```

Comment:
I could go either way on this, but I actually feel that the exception is correct. If someone has explicitly configured that setting but not specified a value, by proto rules, they have specified 0. I think the exception is valid? Thoughts?

/wait-any